pkg/queue/breaker.go

```diff
@@ -133,10 +153,11 @@ func (s *Semaphore) ReduceCapacity(size int32) error {
 // AddCapacity conditionally adds capacity to the semaphore
 // If there are tokens that must be reduced, release them first
 // Otherwise, add tokens to the queue
-func (s *Semaphore) AddCapacity(size int32) {
-	var leftToAdd int32
+// Returns true if all tokens were added, false otherwise
+func (s *Semaphore) AddCapacity(size int32) bool {
```

Comment:
This should also return an error (like ReduceCapacity), so both ways behave the same.